drivers/dma/dma_stm32.c

```diff
@@ -589,20 +790,109 @@ static void dma_stm32_config_irq_1(struct device *dev)
 {
 	struct dma_stm32_data *data = dev->driver_data;
 
-#ifdef DT_INST_1_ST_STM32_DMA_IRQ_0
 	IRQ_INIT(1, 0);
 	IRQ_INIT(1, 1);
 	IRQ_INIT(1, 2);
 	IRQ_INIT(1, 3);
 	IRQ_INIT(1, 4);
 #ifdef DT_INST_1_ST_STM32_DMA_IRQ_5
 	IRQ_INIT(1, 5);
+#endif
+#ifdef DT_INST_1_ST_STM32_DMA_IRQ_6
 	IRQ_INIT(1, 6);
+#endif
 #ifdef DT_INST_1_ST_STM32_DMA_IRQ_7
 	IRQ_INIT(1, 7);
 #endif
-#endif
-#endif
 /* Either 0 or 5 or 7 or 8 channels for DMA1 across all stm32 series. */
 }
-#endif
+#endif /* DT_INST_1_ST_STM32_DMA_IRQ_0 */
+
+#if defined(CONFIG_DMA_STM32_V2) && defined(CONFIG_DMAMUX_STM32)
+static int dmamux_stm32_init(struct device *dev)
+{
+	struct dmamux_stm32_data *data = dev->driver_data;
+	const struct dmamux_stm32_config *config = dev->config->config_info;
+	/* use the same config for dmamux as for dma */
+	struct device *clk =
+		device_get_binding(STM32_CLOCK_CONTROL_NAME);
+
+	if (clock_control_on(clk,
+		(clock_control_subsys_t *) &config->pclken) != 0) {
+		LOG_ERR("clock op failed\n");
+		return -EIO;
+	}
+
+	config->config_irq(dev);
+
+	/* Nb of DMAMUX channels is channel_nb */
+	data->max_channels = config->channel_nb;
+
+	int size_channel =
+		sizeof(struct dmamux_stm32_channel) * data->max_channels;
+	data->channels = k_malloc(size_channel);
+	if (!data->channels) {
+		LOG_ERR("HEAP_MEM_POOL_SIZE is too small");
+		return -ENOMEM;
+	}
+	memset(data->channels, 0, size_channel);
+
+	/* <dmamux-channel-nb> cannot exceed the <dma-channels> per dma */
+	if (config->channel_nb != data->max_channels) {
+		LOG_ERR("Nb of DMAMUX channels is wrong");
+	}
+
+	return 0;
+}
+
+static const struct dma_driver_api dmamux_funcs = {
```

Comment:
If API is not supported, device should be instantiated with DEVICE_INIT.

Reply:
Can you laborate on the reason you have to do this ?

Reply:
The `static const struct dma_driver_api dmamux_funcs = dmamux `
has no api functions by itself, 
but requires a declaration of NULL functions for the DEVICE_API_AND_INIT. 
If using DEVICE_INIT, will make the return a NULL pointer of the `device_get_binding((const char *)"DMAMUX_1");`